pkg/kubelet/cadvisor/util_test.go

```diff
@@ -42,18 +45,57 @@ func TestCapacityFromMachineInfo(t *testing.T) {
 		},
 	}
 
-	// enable the features.HugePages
-	defer utilfeaturetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.HugePages, true)()
+	hgFalseExpected := v1.ResourceList{
+		v1.ResourceCPU:    *resource.NewMilliQuantity(int64(2000), resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(int64(2048), resource.BinarySI),
+	}
+	hgTrueExpected := v1.ResourceList{
+		v1.ResourceCPU:    *resource.NewMilliQuantity(int64(2000), resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(int64(2048), resource.BinarySI),
+		"hugepages-5Ki":   *resource.NewQuantity(int64(51200), resource.BinarySI),
+	}
+	testCapacityFromMachineInfo(t, machineInfo, hgFalseExpected, false)
+	testCapacityFromMachineInfo(t, machineInfo, hgTrueExpected, true)
+}
+
+func testCapacityFromMachineInfo(t *testing.T, inputMI info.MachineInfo, expectedRL v1.ResourceList, fg bool) {
+	defer utilfeaturetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.HugePages, fg)()
+	actual := CapacityFromMachineInfo(&inputMI)
+	if !reflect.DeepEqual(actual, expectedRL) {
+		t.Errorf("when set hugepages %t, got resource list %v, want %v", fg, actual, expectedRL)
+	}
 
-	resourceList := CapacityFromMachineInfo(machineInfo)
+}
 
-	// assert the cpu and memory
-	assert.EqualValues(t, machineInfo.NumCores*1000, resourceList.Cpu().MilliValue(), "unexpected CPU value")
-	assert.EqualValues(t, machineInfo.MemoryCapacity, resourceList.Memory().Value(), "unexpected memory value")
+func TestCrioSocket(t *testing.T) {
+	assert.EqualValues(t, CrioSocket, crio.CrioSocket, "CrioSocket in this package must equal the one in github.com/google/cadvisor/container/crio/client.go")
+}
 
-	// assert the hugepage
-	hugePageKey := int64(5 * 1024)
-	value, found := resourceList[v1helper.HugePageResourceName(*resource.NewQuantity(hugePageKey, resource.BinarySI))]
-	assert.True(t, found, "hugepage not found")
-	assert.EqualValues(t, hugePageKey*10, value.Value(), "unexpected hugepage value")
+func TestUsingLegacyCadvisorStats(t *testing.T) {
+	testcases := []struct {
+		name            string
+		runtime         string
+		runtimeEndpoint string
+		expectedLinux   bool
+		expectedWindows bool
+	}{
+		{"case one: use legacy for both Linux and Windows", "docker", "/var/run/crio/crio.sock", true, true},
```

Comment:
nit: describe the case you are testing, not the outcome.  E.g. Using dockershim.  I also don't think runtime == docker and setting the endpoint to crio would be a valid configuration.  The alternative to using docker is using `kubetypes.RemoteContainerRuntime`